pyfftw/interfaces/_utils.py

```diff
@@ -97,24 +101,25 @@ def _Xfftn(a, s, axes, overwrite_input, planner_effort,
         # take the transform of the input array! (in general, we have
         # to assume that the input array will be destroyed during 
         # planning).
-        a_copy = a.copy()
+        if overwrite_input:    # Note: reload_after_transform is not True...
+            planner_args = (a, s, axes) + args
+            a_copy = a.copy()  # ... hence this is not an additional copy
+        else:
+            planner_args = (numpy.empty_like(a), s, axes) + args
```

Comment:
This should possibly use `numpy.lib.stride_tricks.as_strided` but one needs to be careful with this to make sure that it is applied to a large-enough chunk of memory.  One possibility might be to inspect `a` and make copy of the original array, then apply the `as_strided`.  However, this could cause problems if the original array is huge since `a` might be a small view.  In this case, the original code - transforming on a copy and then restoring it - would be preferable and might require one more special case.

That being said, I still can't find a way that this code actually fails.  Perhaps it is just a performance issue.